mlir/lib/Dialect/Linalg/Transforms/TransposeMatmul.cpp

```diff
@@ -88,6 +88,14 @@ FailureOr<Operation *>
 mlir::linalg::transposeBatchMatmul(RewriterBase &rewriter,
                                    linalg::BatchMatmulOp batchMatmulOp,
                                    bool transposeLHS) {
+  // Check to not let go the batch_matmul with extended semantic, through this
+  // transform.
+  if (batchMatmulOp.hasUserDefinedMaps()) {
+    return rewriter.notifyMatchFailure(
+        batchMatmulOp,
+        "only batch_matmul ops with non-extended semantics are supported");
```

Comment:
Either `linalg.batch_matmul` or skip altogether. IIRC, the Op will be printed anyway. Also:
```suggestion
        "user-defined maps are not supported");
```

Reply:
Done